submarine-server/server-rpc/src/main/java/org/apache/submarine/server/rpc/SubmarineRpcServer.java

```diff
@@ -109,11 +116,38 @@ private static ClientContext getClientContext(RpcContext rpcContext) {
     ClientContext clientContext = new ClientContext();
     clientContext.setYarnConfig(conf);
     mergeSubmarineConfiguration(clientContext.getSubmarineConfig(), rpcContext);
+    ClassLoader classLoader = new URLClassLoader(constructUrlsFromClasspath("../lib/submitter"));
+
     RuntimeFactory runtimeFactory = RuntimeFactory.getRuntimeFactory(
-        clientContext);
+        clientContext, classLoader);
     clientContext.setRuntimeFactory(runtimeFactory);
     return clientContext;
   }
+  private static URL[] constructUrlsFromClasspath(String classpath) {
+    List<URL> urls = new ArrayList<>();
+    for (String path : classpath.split(File.pathSeparator)) {
+      if (path.endsWith("/*")) {
```

Comment:
Maybe no recursive loading jar?

Reply:
It is not a recurisive loading jar. Only the jars in the folder will be loaded. The jars in the subfolder will not.